From d267c5aa3d0fe4960165a1e1c00a840eef8b7d00 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Fri, 3 Oct 2014 19:17:56 -0400
Subject: [PATCH] core/namespace: remove invalid check

dir cannot be NULL here, because it was allocated with alloca.

CID #1237768.
---
 src/core/namespace.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/core/namespace.c b/src/core/namespace.c
index f86092f6b1..c2215090a9 100644
--- a/src/core/namespace.c
+++ b/src/core/namespace.c
@@ -263,11 +263,8 @@ fail:
         if (devmqueue)
                 umount(devmqueue);
 
-        if (dev) {
-                umount(dev);
-                rmdir(dev);
-        }
-
+        umount(dev);
+        rmdir(dev);
         rmdir(temporary_mount);
 
         return r;
